instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/instrumentation.rb

```diff
@@ -31,26 +31,24 @@ def gem_version
         end
 
         def patch
-          # The original approach taken here was to patch each individual module of interest.
-          # However the patches are applied too late in some applications and as a result the
-          # Active Record models will not have the instrumentation patches applied.
-          # Prepending the ActiveRecord::Base class is more consistent in applying
-          # the patches regardless of initialization order.
-          #
-          # Modules to prepend to ActiveRecord::Base are still grouped by the source
-          # module that they are defined in.
-          # Example: Patches::PersistenceClassMethods refers to https://github.com/rails/rails/blob/v6.1.0/activerecord/lib/active_record/persistence.rb#L10
-          ::ActiveRecord::Base.prepend(Patches::Querying)
-          ::ActiveRecord::Base.prepend(Patches::Persistence)
-          ::ActiveRecord::Base.prepend(Patches::PersistenceClassMethods)
-          ::ActiveRecord::Base.prepend(Patches::PersistenceInsertClassMethods)
-          ::ActiveRecord::Base.prepend(Patches::TransactionsClassMethods)
-          ::ActiveRecord::Base.prepend(Patches::Validations)
-
-          ::ActiveRecord::Relation.prepend(Patches::RelationPersistence)
+          ActiveSupport.on_load(:active_record) do
+            # Modules to prepend to ActiveRecord::Base are grouped by the source
+            # module that they are defined in as they are included into ActiveRecord::Base
+            # Example: Patches::PersistenceClassMethods refers to https://github.com/rails/rails/blob/v6.1.0/activerecord/lib/active_record/persistence.rb#L10
+            #   which is included into ActiveRecord::Base in https://github.com/rails/rails/blob/914caca2d31bd753f47f9168f2a375921d9e91cc/activerecord/lib/active_record/base.rb#L283
+            ::ActiveRecord::Base.prepend(Patches::Querying)
```

Comment:
It occurs to me now that none of these use active support concerns.

Is there any issue related to code reloading that would be alleviated by using concerns?

Reply:
None of the concerns have lazy-load-hooks, so they end up just being loaded at the same time.

I had to read the original comment like 5 times, but I think what it means is that the Rails code being patched here has been extracted into a concern by Rails, and OTel is patching it, with a similar naming convention, when it's included back into `ActiveRecord::Base`.

Also `Concern` is waaaay too overloaded as a word cause it can both mean "includes ActiveSupport::Concern which has some nice helper methods for extraction/inclusion" but also just like "a module with some behavior that's been extracted that's intended to be included back".  ¯\\\_(ツ)_/¯